### PR TITLE
fix js2py for object -> dict

### PR DESCRIPTION
**Pull request by jstoebel**
- require `types` inside of hash function
 - use `__setitem__` in js2py rather than tacking on a new attr

### CODE REVIEW

File: batavia/builtins/hash.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 var exceptions = require('../core').exceptions
 var type_name = require('../core').type_name
-var types = require('../types')
 
 function hash(args, kwargs) {
+    var types = require('../types')
```
Comment (dimaqq) [inline batavia/builtins/hash.js]: Is this change intended?

Node.js advises against `require` inside function body:
https://github.com/alanjames1987/Node.js-Best-Practices/issues/3

Comment (freakboy3742) [inline batavia/builtins/hash.js]: Node.js may well recommend that - but have they provided an alternative way to resolve circular dependencies in imports?

Comment (dimaqq) [inline batavia/builtins/hash.js]: It's a fair point that something must be done.
However require on each call to `hash()` seems like a crazy penalty, doesn't it?

I'm assuming it's going to be used a lot.
P.S. I don't see `hash` being used by `types` anyway...

Comment (jstoebel) [inline batavia/builtins/hash.js]: @dimaqq I moved `var types = require('../types')` because I was otherwise unable to access `types.isinstance` inside of the `hash` function

**Comment by jstoebel** [comment]
@freakboy3742 yes, I will put together some tests for `js2py`
